nautobot/extras/models/approvals.py

```diff
@@ -138,6 +141,48 @@ def __str__(self):
         """Stringify instance."""
         return f"{self.approval_workflow.name}: {self.object_under_review} ({self.current_state})"
 
+    @property
+    def current_stage(self):
+        """
+        Return the current stage of the workflow.
+        1. Return the stage that is denied if there is a stage that has been denied.
+        2. Return the first stage that is pending, if there is no stage that has been approved or denied.
+        3. Return the stage that is pending that is immediately after the last stage that has been approved.
+        4. Return None, if all stages are approved.
+        """
+        # Check if there is a stage that is denied
+        denied_stage = self.approval_workflow_stage_instances.filter(state=ApprovalWorkflowStateChoices.DENIED)
+        if denied_stage.exists():
+            return denied_stage.first()
+        # Get the pending stage with the lowest weight
+        pending_stages = self.approval_workflow_stage_instances.filter(
+            state=ApprovalWorkflowStateChoices.PENDING
+        ).order_by("approval_workflow_stage__weight")
+        if pending_stages.exists():
+            return pending_stages.first()
```

Comment:
Actually, could this be simplified to just be "first stage that isn't Approved"? Something like:

```py
first_nonapproved_stage = self.approval_workflow_stage_instances.filter(
    state__n=ApprovalWorkflowStateChoices.APPROVED
).first()
return first_nonapproved_stage
```

Reply:
Nice!